src/components/select/simple-select/simple-select.stories.tsx

```diff
@@ -854,3 +854,52 @@ export const SelectWithDynamicallyAddedOption: Story = () => {
   );
 };
 SelectWithDynamicallyAddedOption.storyName = "Dynamically Adding Options";
+
+export const ComplexCustomChildren: Story = () => {
+  return (
+    <Box height={220}>
+      <Select
+        mb={0}
+        key="key"
+        id="id"
+        label="Select"
+        aria-label="aria label"
+        name="name"
+        value="value"
+        isLoading={false}
+        readOnly={false}
+        placeholder="placeholder"
+        onChange={() => {}}
+        onOpen={() => {}}
+        onListScrollBottom={() => {}}
+      >
+        <Option>
+          <Box
+            width="100%"
+            display="flex"
+            alignItems="center"
+            justifyContent="center"
+            mt={2}
+            mb={3}
+            flexDirection="column"
+          >
+            <Box display="flex" mx={2}>
+              <Icon type="error" color="errorRed" />
+              <Box ml={1} width="100%">
+                <Box mb={1}>
+                  <Typography variant="b" color="errorRed">
+                    Something went wrong
+                  </Typography>
+                </Box>
+                <Typography variant="p" color="errorRed" mb={0}>
+                  We couldn't load the data. Please try again later.
+                </Typography>
+              </Box>
+            </Box>
+          </Box>
+        </Option>
```

Comment:
**comment**: I have a few thoughts regarding the idea of making `Option` a composable slot - particularly regarding the composition of non-selectable content.

While it’s true that this content wouldn’t be selectable, due to the [custom logic we have in `SelectList`](https://github.com/Sage/carbon/blob/master/src/components/select/__internal__/select-list/select-list.component.tsx#L340-L368), the way it’s rendered inside a `<li>` tag within a `<ul>` list isn't semantically correct:

```html
<!-- Error message is technically an option -->
<ul>
  <li aria-setsize="1" aria-posinset="1">
    <div>Error message</div>
  </li>
</ul>
```

Secondly, even though this content isn’t selectable, the fact that it can’t be navigated to via keyboard could pose accessibility challenges for screen readers, as they won’t be able to access this information.

Thirdly, if we allow non-selectable content to be rendered in the dropdown popup but outside the list, like this:

```html
<!-- Correct rendering for non-selectable content -->
<ul>
</ul>
<div>Error message</div>
```

We would need to make sure that the content can still be accessed by keyboard users, without the list closely prematurely.

To ensure clarity, it might be useful to document that Option isn’t intended for non-option content, while also encouraging consumers to explore alternative UX patterns for handling errors in dynamic Select lists.

Reply:
I think I'll move the story to a test story for QA and then remove it from public docs but add some text around  how not setting a value on the Option means it's content is non-interactive and that in general it's not recommended etc

Reply:
Yeah, I appreciate we don't want to make too many breaking changes at this stage. I’m happy with making this a test story for QA, and adding a disclaimer note about its intended use for now 👍🏼